drv/cosmo-hf/src/apob.rs

```diff
@@ -0,0 +1,124 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tools to extract the APOB location from an AMD ROM
+//!
+//! For details, see AMD document 57299; tables and sections in this code refer
+//! to Rev. 2.0 February 2025.
+
+use crate::hf::ServerImpl;
+use drv_hf_api::HfError;
+use userlib::UnwrapLite;
+use zerocopy::{FromBytes, Immutable, IntoBytes};
+
+/// Embedded firmware structure (Table 3)
+///
+/// Only relevant fields are included here.
+#[derive(FromBytes, Immutable, IntoBytes)]
+#[repr(C)]
+pub struct Efs {
+    signature: u32,
+    _padding1: [u8; 16],
+    psp_dir_offset: u32,
+    _padding2: [u8; 16],
+    bios_dir_offset: u32,
+}
+
+const EFS_SIGNATURE: u32 = 0x55aa55aa;
+const BHD_DIR_COOKIE: u32 = 0x44484224; // $BHD
+const APOB_NV_COPY: u8 = 0x63; // Table 29
+
+/// BIOS Directory Table Header (Table 17)
+#[derive(FromBytes, Immutable, IntoBytes)]
+#[repr(C)]
+pub struct BhdDir {
+    cookie: u32,
+    checksum: u32,
+    num: u32,
+    info: u32,
+}
+
+/// BIOS Directory Table Entry (Table 18)
+#[derive(FromBytes, Immutable, IntoBytes)]
+#[repr(C)]
+pub struct DirEntry {
+    entry_type: u8,
+    region_type: u8,
+    _unused1: u8,
+    _unused2: u8,
```

Comment:
nit, take it or leave it: just to be consistent with the `Efs` struct in this module, why not:
```suggestion
    _padding [u8; 2],
```

Reply:
Fixed!